TodosApis.Dapper.IntegrationTests/TodosApiIntegration.cs

```diff
@@ -0,0 +1,573 @@
+
+        var todos = await httpClient.GetFromJsonAsync<List<Todo>>("/todos");
+
+        Assert.Empty(todos);
+    }
+
+    [Fact]
+    public async Task GetTodos_IsNotEmpty_ReturnsOk()
+    {
+        using var application = new TodosApplication();
+
+        var httpClient = application.CreateClient();
+
+        var newTodo = new NewTodo
+        {
+            Title = "Create Integration Tests"
+        };
+
+        var response = await httpClient.PostAsJsonAsync("/todos", newTodo);
+
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        var todos = await httpClient.GetFromJsonAsync<List<Todo>>("/todos");
+
+        Assert.NotEmpty(todos);
+    }
+
+    [Fact]
+    public async Task PostTodo_Returns_Created()
+    {
+        using var application = new TodosApplication();
+
+        var httpClient = application.CreateClient();
+
+        var newTodo = new NewTodo
+        {
+            Title = "Create Integration Tests"
+        };
+        
+        var response = await httpClient.PostAsJsonAsync("/todos", newTodo);
+
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        var todos = await httpClient.GetFromJsonAsync<List<Todo>>("/todos");
+
+        var todo = Assert.Single(todos);
+        Assert.Equal(newTodo.Title, todo.Title);
+        Assert.False(todo.IsComplete);
+    }
+    
+    [Fact]
+    public async Task PostTodo_Returns_BadRequest()
+    {
+        using var application = new TodosApplication();
+
+        var httpClient = application.CreateClient();
+
+        var newTodo = new NewTodo
+        {
+            Title = null
+        };
+
+        var response = await httpClient.PostAsJsonAsync("/todos", newTodo);
+
+        Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+        var todos = await httpClient.GetFromJsonAsync<List<Todo>>("/todos");
+
+        Assert.Empty(todos);
+    }
+
+    [Fact]
+    public async Task UpdateTodo_Returns_NoContent()
+    {
+        await using var application = new TodosApplication();
+
+        var httpClient = application.CreateClient();
+
+        var newTodoExpected = new NewTodo
+        {
+            Title = "Create Integration Tests"
+        };
+
+        var newTitle = "Title Updated";
+
+        var response = await httpClient.PostAsJsonAsync("/todos", newTodoExpected);
+
+        Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+        var todos = await httpClient.GetFromJsonAsync<List<Todo>>("/todos");
+
+        var todo = Assert.Single(todos);
+        Assert.Equal(newTodoExpected.Title, todo.Title);
+        Assert.False(todo.IsComplete);
+
+        todo.Title = newTitle;
+
+        response = await httpClient.PutAsJsonAsync($"/todos/{todo.Id}", todo);
+
+        Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
+
+        todo = await httpClient.GetFromJsonAsync<Todo>($"/todos/{todo.Id}");
+
+        Assert.Equal(newTitle, todo.Title);
```

Comment:
Fix all compiler nullability warnings:
```suggestion
        Assert.Equal(newTitle, todo?.Title);
```